src/transaction_builder.js

```diff
@@ -108,7 +108,7 @@ function expandOutput (script, scriptType, ourPubKey) {
 
   var scriptChunks = bscript.decompile(script)
   if (!scriptType) {
-    scriptType = bscript.classifyOutput(scriptChunks)
+    scriptType = bscript.classifyOutput(script)
```

Comment:
Going to rever this... it was done specifically as an optimization.  Will comment that though.